### PR TITLE
Fix: Storage reset, default settings, and stale state on sign-out / guest login

### PR DESCRIPTION
**Pull request by RasmusAChr**
# Overview
This PR started as a fix for clearing local storage on sign-out. During *investigation*, I found out that several parts of the extension were not reading from storage at all, but were falling back to hardcoded default values defined in `defaultSettingValues.js`, which meant that clearing storage alone wasn't enough to guarantee a clean state (primarily for guest login).

Addresses issue #60.

> [!important]
> I've tried my best to test all different cases were it could go wrong. However, please try out some edge cases to see if everything is still working as intended.

# Changelog

### Storage reset on sign-out and guest login
`clearSessionLocally` now calls `storage.clearStorage()` to clear everything except for the theme, which should be dark by default. When signing in as a guest (or when `loadDefaults: true` is passed), it also writes the default settings explicitly into storage so the app always reads consistent values from the start.

### Default settings are now written on guest login
A new `loadDefaultSettingsLocally` function in `SetUser.svelte` writes operating hours, learning time, session duration, and reward time to storage using the default values, so the app never silently falls back to *'in-memory'* constants.

### Redirection is enabled correctly after login
When loading the extension for the first time, nothing is stored in local storage anymore. Now, after a successful login or guest sign-in, `shouldRedirect` is set to `true` and the redirection toggle is explicitly enabled.

### `clearStorage` is now awaitable
`storage.clearStorage()` previously called and forgot. It now returns a promise from `browser.storage.local.clear()` so callers can await it correctly.

### Installation setup simplified
Removed redundant explicit initialization calls from `installationSetup` and `setup` in `setupService.js`. Default values are now the responsibility of storage getters and the new `loadDefaultSettingsLocally` flow, avoiding double-initialisation.

### CODE REVIEW

File: src/handlers/apiHandler.js [hunk 1]
```diff
@@ -35,16 +33,6 @@ export async function handleApiMessage(message) {
...
-      try {
-        const syncResult = await fetchAndSyncSettings();
-        if (!syncResult.ok) {
-          console.warn("[Settings] Could not sync settings on login:", syncResult.message);
-        }
-      } catch (e) {
-        console.warn("[Settings] fetchAndSyncSettings crashed: ", e);
-      }
-      // Has to restart listener, as when the user logs in, they will have an empty list of timewasting sites.
-      // Without this, it would never restart and actually check on the sites the user has added. It would check on the "old" list, which most likely was empty
```
Comment (Nikolaj787) [inline src/handlers/apiHandler.js:47]: Very minor detail as it can be discussed whether or not the code comments is needed. But since the code that the comments refers to is still in place, we should probably not as a default remove such comments 😅

**Review by Nikolaj787 — APPROVED** [review]
I didn't encounter any issues when testing it, so nicely done!

**Review by niko391a — APPROVED** [review]
